tests/test_tnstate.py

```diff
@@ -481,13 +483,19 @@ def test_circ_approx_explicit_ttn(circuit: Circuit) -> None:
     random.seed(1)
 
     with CuTensorNetHandle() as libhandle:
+        # Finite gate fidelity
+        # Check for TTNxGate
+        cfg = Config(truncation_fidelity=0.99)
+        ttn_gate = simulate(libhandle, circuit, SimulationAlgorithm.TTNxGate, cfg)
+        assert np.isclose(ttn_gate.get_fidelity(), 0.69, atol=1e-2)
+        assert ttn_gate.is_valid()
+        assert np.isclose(ttn_gate.vdot(ttn_gate), 1.0, atol=cfg._atol)
+
         # Fixed virtual bond dimension
         # Check for TTNxGate
         cfg = Config(chi=120, leaf_size=3)
         ttn_gate = simulate(libhandle, circuit, SimulationAlgorithm.TTNxGate, cfg)
-        for g in circuit.get_commands():
-            ttn_gate.apply_gate(g)

```

Comment:
These two lines of code are applying all gates in the circuit _again_ for some reason. I suppose it's because on the first version I was not yet running `simulate` and instead applying the gates like this and when I implemented `simulate` I missed removing this for loop from this test. What was happening is that the circuit being simulated was "doubled", pasting the end of the first copy to the beginning of the next. So, of course, the fidelity of this longer depth was worse.